advanced/src/hashTable.js

```diff
@@ -0,0 +1,227 @@
+function Node(key, value) {
+  this.key = key
+  this.value = value
+  this.next = null
+}
+
+
+const LinkedList = class LinkedList {
+  constructor() {
+    this.length = 0
+    this.head = null
+  }
+
+  insert(key, value) {
+    let node = new Node(key, value)
+    let currentNode = this.head
+
+    if (!currentNode) {
+      this.head = node
+      this.length++
+      return node
+    }
+    while (currentNode.next) {
+      currentNode = currentNode.next
+    }
+    currentNode.next = node
+    this.length += 1
+    return node
+  }
+
+//regular find
+  find(key) {
+    var currentNode = this.head
+    if (!currentNode) {
+      return false
+    }
+    while (currentNode) {
+      if (currentNode.key === key){
+        return currentNode
+      }
+    } return false
+  }
+
+//find but return the value
+  findButReturnValue(key) {
+    const node = this.find(key)
+    return node
+      ? node.value
+      : null
+  }
+
+//see if linked list contains key
+  has(key) {
+    return !!this.find(key)
+  }
+
+//clears my linked list
+  clear() {
+    this.length = 0
+    this.head = null
+  }
+
+//finds the node whose next value is the target node
+  findByNext(key) {
+    let currentNode = this.head
+
+    if(!currentNode) {
+      return null
+    } else if (!currentNode.next && currentNode.key === key){
+      return this.clear()
```

Comment:
This seems... unintentional?  You clear the list if you find that the head element contains the search key, and doesn't have a next element?